### ISSUE TITLE
Build numpy: '['/kivy/dist/hostpython3/bin/python', '-m', 'cython', '-3', '--fast-fail', '-o', '_philox.c', '_philox.pyx']' returned non-zero exit status 1.'

### ISSUE THREAD

**Issue by krs1w** [kivy/kivy-ios#835]
**Versions**

* Python : 3.10.10 / 3.10.12 (Installed globally with pyenv)
* MacOS version : Ventura 13.4.1
* XCode Version : 14.3.1
* Cython version : 3.0.0 (Installed through dependencies)

**Describe the bug**
My kivy-setup for building python and numpy was working a few days ago, but fails as of today.
Numpy shows some deprecation warnings and ultimately fails with 
`'['/Users/fprefect/Developer/kivy/dist/hostpython3/bin/python', '-m', 'cython', '-3', '--fast-fail', '-o', '_philox.c', '_philox.pyx']' returned non-zero exit status 1.`
I have reproduced this error with a pip installed kivy-ios and by manual checkout, on M1 Mac mini and Intel MacBook. 
All prior recipes in the build run work flawlessly (hostopenssl, hostpython3, libffi, openssl, python3) producing libcrypto.a, libffi.a, libpython3.a, libssl.a.
The log output looks kind of cut off at the end but this is the complete output of the console.
I did cut some of the numpy log naming all files extracted but if you need any more logs or infos I'm happy to provide :)  

**To Reproduce**
- python3.10 -m venv venv
- . ./venv/bin/activate
- pip3 install kivy-ios
- xcode-select --install
- brew install autoconf automake libtool pkg-config
- toolchain build numpy

**Expected behavior**
Working build, producing dist/lib/libnumpy.a

**Logs** 
- pip3 install kivy-ios
```
Collecting kivy-ios
  Using cached kivy_ios-2023.5.21-py3-none-any.whl (1.1 MB)
Collecting sh
  Using cached sh-2.0.4-py3-none-any.whl (38 kB)
Collecting cookiecutter
  Using cached cookiecutter-2.2.3-py3-none-any.whl (39 kB)
Collecting requests
  Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Collecting pbxproj
  Using cached pbxproj-3.5.0-py3-none-any.whl (40 kB)
Collecting Cython
  Using cached Cython-3.0.0-py2.py3-none-any.whl (1.2 MB)
Collecting Pillow
  Using cached Pillow-10.0.0-cp310-cp310-macosx_11_0_arm64.whl (3.1 MB)
Collecting click<9.0.0,>=7.0
  Using cached click-8.1.6-py3-none-any.whl (97 kB)
Collecting Jinja2<4.0.0,>=2.7
  Using cached Jinja2-3.1.2-py3-none-any.whl (133 kB)
Collecting pyyaml>=5.3.1
  Using cached PyYAML-6.0.1-cp310-cp310-macosx_11_0_arm64.whl (169 kB)
Collecting python-slugify>=4.0.0
  Using cached python_slugify-8.0.1-py2.py3-none-any.whl (9.7 kB)
Collecting arrow
  Using cached arrow-1.2.3-py3-none-any.whl (66 kB)
Collecting binaryornot>=0.4.4
  Using cached binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting certifi>=2017.4.17
  Using cached certifi-2023.5.7-py3-none-any.whl (156 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.2.0-cp310-cp310-macosx_11_0_arm64.whl (124 kB)
Collecting urllib3<3,>=1.21.1
  Using cached urllib3-2.0.3-py3-none-any.whl (123 kB)
Collecting docopt
  Using cached docopt-0.6.2.tar.gz (25 kB)
  Preparing metadata (setup.py) ... done
Collecting openstep-parser>=1.5.1
  Using cached openstep_parser-1.5.4-py3-none-any.whl (4.5 kB)
Collecting chardet>=3.0.2
  Using cached chardet-5.1.0-py3-none-any.whl (199 kB)
Collecting MarkupSafe>=2.0
  Using cached MarkupSafe-2.1.3-cp310-cp310-macosx_10_9_universal2.whl (17 kB)
Collecting text-unidecode>=1.3
  Using cached text_unidecode-1.3-py2.py3-none-any.whl (78 kB)
Collecting python-dateutil>=2.7.0
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: text-unidecode, openstep-parser, docopt, urllib3, six, sh, pyyaml, python-slugify, Pillow, pbxproj, MarkupSafe, idna, Cython, click, charset-normalizer, chardet, certifi, requests, python-dateutil, Jinja2, binaryornot, arrow, cookiecutter, kivy-ios
  DEPRECATION: docopt is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559
  Running setup.py install for docopt ... done
Successfully installed Cython-3.0.0 Jinja2-3.1.2 MarkupSafe-2.1.3 Pillow-10.0.0 arrow-1.2.3 binaryornot-0.4.4 certifi-2023.5.7 chardet-5.1.0 charset-normalizer-3.2.0 click-8.1.6 cookiecutter-2.2.3 docopt-0.6.2 idna-3.4 kivy-ios-2023.5.21 openstep-parser-1.5.4 pbxproj-3.5.0 python-dateutil-2.8.2 python-slugify-8.0.1 pyyaml-6.0.1 requests-2.31.0 sh-2.0.4 six-1.16.0 text-unidecode-1.3 urllib3-2.0.3

[notice] A new release of pip available: 22.3.1 -> 23.2
[notice] To update, run: pip install --upgrade pip
```
- toolchain build numpy
```
[INFO    ] Download numpy
[INFO    ] Downloading https://pypi.python.org/packages/source/n/numpy/numpy-1.24.2.tar.gz
[DEBUG   ] New State: numpy.download at 2023-07-18 23:44:43.159955
[INFO    ] Extract numpy
[INFO    ] Extract numpy for x86_64
[INFO    ] Extract /Users/fprefect/Developer/kivy/.cache/numpy-numpy-1.24.2.tar.gz into /Users/fprefect/Developer/kivy/build/numpy/x86_64
[INFO    ] Running Shell: /usr/bin/tar ('-C', '/Users/fprefect/Developer/kivy/build/numpy/x86_64', '-xv', '-z', '-f', '/Users/fprefect/Developer/kivy/.cache/numpy-numpy-1.24.2.tar.gz') {'_iter': True, '_out_bufsize': 1, '_err_to_out': True}
[DEBUG   ] x numpy-1.24.2/
[DEBUG   ] x numpy-1.24.2/.coveragerc

[. . .] 

[DEBUG   ] x numpy-1.24.2/tox.ini
[DEBUG   ] x numpy-1.24.2/versioneer.py
[DEBUG   ] New State: numpy.extract at 2023-07-18 23:44:43.838770
[INFO    ] Install_hostpython_prerequisites numpy
[INFO    ] Executing pip for hostpython with: ['install', 'Cython']
[INFO    ] Running Shell: /Users/fprefect/Developer/kivy/dist/hostpython3/bin/pip3 ('install', 'Cython') {'_iter': True, '_out_bufsize': 1, '_err_to_out': True}
[DEBUG   ] Collecting Cython
[DEBUG   ]   Using cached Cython-3.0.0-py2.py3-none-any.whl (1.2 MB)
[DEBUG   ] Installing collected packages: Cython
[DEBUG   ] Successfully installed Cython-3.0.0
[DEBUG   ] 
[DEBUG   ] [notice] A new release of pip available: 22.3.1 -> 23.2
[DEBUG   ] [notice] To update, run: /Users/fprefect/Developer/kivy/dist/hostpython3/bin/python3.10 -m pip install --upgrade pip
[DEBUG   ] New State: numpy.install_hostpython_prerequisites at 2023-07-18 23:44:46.033334
[INFO    ] Build_all numpy
[INFO    ] Build numpy for x86_64, arm64 (filtered)
[INFO    ] Build numpy
[INFO    ] Prebuild numpy for x86_64
[INFO    ] Apply patch skip-math-test.patch
[INFO    ] Apply patch duplicated_symbols.patch
[INFO    ] Build numpy for x86_64
[INFO    ] Running Shell: /Users/fprefect/Developer/kivy/dist/hostpython3/bin/python ('setup.py', 'build_ext', '-g') {'_env': {'CC': '/var/folders/k8/h2f5jsvx7pd6dzrjjn58ph0r0000gn/T/tmp3xscqtl5 -O3 -miphoneos-version-min=9.0 -I/Users/fprefect/Developer/kivy/dist/include/x86_64/ffi -I/Users/fprefect/Developer/kivy/dist/include/x86_64/openssl -I/Users/fprefect/Developer/kivy/dist/include/common/numpy -I/Users/fprefect/Developer/kivy/dist/include/x86_64', 'CXX': '/var/folders/k8/h2f5jsvx7pd6dzrjjn58ph0r0000gn/T/tmp629f_x6f', 'AR': '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ar', 'LD': '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld', 'OTHER_CFLAGS': '-I/Users/fprefect/Developer/kivy/dist/include/x86_64/ffi -I/Users/fprefect/Developer/kivy/dist/include/x86_64/openssl -I/Users/fprefect/Developer/kivy/dist/include/common/numpy -I/Users/fprefect/Developer/kivy/dist/include/x86_64', 'OTHER_LDFLAGS': '-L/Users/fprefect/Developer/kivy/dist/lib', 'CFLAGS': '-O3 -miphoneos-version-min=9.0 -I/Users/fprefect/Developer/kivy/dist/include/x86_64/ffi -I/Users/fprefect/Developer/kivy/dist/include/x86_64/openssl -I/Users/fprefect/Developer/kivy/dist/include/common/numpy -I/Users/fprefect/Developer/kivy/dist/include/x86_64', 'LDFLAGS': '-arch x86_64 -L/Users/fprefect/Developer/kivy/dist/lib -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk/usr/lib -miphoneos-version-min=9.0', 'KIVYIOSROOT': '/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios', 'IOSSDKROOT': '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.4.sdk', 'CUSTOMIZED_OSX_COMPILER': 'True', 'LDSHARED': '/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/tools/liblink', 'ARM_LD': '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld', 'ARCH': 'x86_64', 'NPY_BLAS_ORDER': '', 'NPY_LAPACK_ORDER': ''}, '_iter': True, '_out_bufsize': 1, '_err_to_out': True}
[DEBUG   ] Running from numpy source directory.
[DEBUG   ] /Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/setup.py:67: DeprecationWarning: 
[DEBUG   ] 
[DEBUG   ]   `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
[DEBUG   ]   of the deprecation of `distutils` itself. It will be removed for
[DEBUG   ]   Python >= 3.12. For older Python versions it will remain present.
[DEBUG   ]   It is recommended to use `setuptools < 60.0` for those Python versions.
[DEBUG   ]   For more details, see:
[DEBUG   ]     https://numpy.org/devdocs/reference/distutils_status_migration.html 
[DEBUG   ] 
[DEBUG   ] 
[DEBUG   ]   import numpy.distutils.command.sdist
[DEBUG   ] Cythonizing sources
[DEBUG   ] Processing numpy/random/_bounded_integers.pxd.in
[DEBUG   ] Processing numpy/random/_philox.pyx
[DEBUG   ] warning: _philox.pyx:14:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
[DEBUG   ] 
[DEBUG   ] Error compiling Cython file:
[DEBUG   ] ------------------------------------------------------------
[DEBUG   ] ...
[DEBUG   ]             self.rng_state.ctr.v[i] = counter[i]
[DEBUG   ] 
[DEBUG   ]         self._reset_state_variables()
[DEBUG   ] 
[DEBUG   ]         self._bitgen.state = <void *>&self.rng_state
[DEBUG   ]         self._bitgen.next_uint64 = &philox_uint64
[DEBUG   ]                                    ^
[DEBUG   ] ------------------------------------------------------------
[DEBUG   ] 
[DEBUG   ] _philox.pyx:190:35: Cannot assign type 'uint64_t (*)(void *) except? -1 nogil' to 'uint64_t (*)(void *) noexcept nogil'
[DEBUG   ] Traceback (most recent call last):
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/tools/cythonize.py", line 205, in <module>
[DEBUG   ]     main()
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/tools/cythonize.py", line 201, in main
[DEBUG   ]     find_process_files(root_dir)
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/tools/cythonize.py", line 192, in find_process_files
[DEBUG   ]     process(root_dir, fromfile, tofile, function, hash_db)
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/tools/cythonize.py", line 158, in process
[DEBUG   ]     processor_function(fromfile, tofile)
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/tools/cythonize.py", line 47, in process_pyx
[DEBUG   ]     subprocess.check_call(
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/dist/hostpython3/lib/python3.10/subprocess.py", line 369, in check_call
[DEBUG   ]     raise CalledProcessError(retcode, cmd)
[DEBUG   ] subprocess.CalledProcessError: Command '['/Users/fprefect/Developer/kivy/dist/hostpython3/bin/python', '-m', 'cython', '-3', '--fast-fail', '-o', '_philox.c', '_philox.pyx']' returned non-zero exit status 1.
[DEBUG   ] Traceback (most recent call last):
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/setup.py", line 479, in <module>
[DEBUG   ]     setup_package()
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/setup.py", line 458, in setup_package
[DEBUG   ]     generate_cython()
[DEBUG   ]   File "/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/setup.py", line 261, in generate_cython
[DEBUG   ]     raise RuntimeError("Running cythonize failed!")
[DEBUG   ] RuntimeError: Running cythonize failed!
Exception in thread background thread for pid 73573:
Traceback (most recent call last):
  File "/Users/fprefect/.pyenv/versions/3.10.10/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/Users/fprefect/.pyenv/versions/3.10.10/lib/python3.10/threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 1655, in wrap
    fn(*rgs, **kwargs)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 2657, in background_thread
    handle_exit_code(exit_code)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 2348, in fn
    return self.command.handle_command_exit_code(exit_code)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 839, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /Users/fprefect/Developer/kivy/dist/hostpython3/bin/python setup.py build_ext -g

  STDOUT:
Running from numpy source directory.
/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/setup.py:67: DeprecationWarning: 

  `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
  of the deprecation of `distutils` itself. It will be removed for
  Python >= 3.12. For older Python versions it will remain present.
  It is recommended to use `setuptools < 60.0` for those Python versions.
  For more details, see:
    https://numpy.org/devdocs/reference/distutils_status_migration.html 


  import numpy.distutils.command.sdist
Cythonizing sources
Processing numpy/random/_bounded_integers.pxd.in
Processing numpy/random/_philox.pyx
warning: _philox.pyx:14:0: The 'DEF' statement is deprecated and will be removed in a future Cyt... (2281 more, please see e.stdout)

  STDERR:

Traceback (most recent call last):
  File "/Users/fprefect/Developer/kivy/venv/bin/toolchain", line 8, in <module>
    sys.exit(main())
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 1564, in main
    ToolchainCL()
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 1308, in __init__
    getattr(self, args.command)()
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 1377, in build
    build_recipes(args.recipe, ctx)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 1142, in build_recipes
    recipe.execute()
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 704, in execute
    self.build_all()
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 74, in _cache_execution
    f(self, *args, **kwargs)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 803, in build_all
    self.build(arch)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 74, in _cache_execution
    f(self, *args, **kwargs)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 790, in build
    self.build_arch(arch)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/recipes/numpy/__init__.py", line 36, in build_arch
    super().build_arch(arch)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 1095, in build_arch
    shprint(hostpython, "setup.py", "build_ext", "-g",
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/kivy_ios/toolchain.py", line 56, in shprint
    for line in cmd:
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 895, in __next__
    self.wait()
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 812, in wait
    self.handle_command_exit_code(exit_code)
  File "/Users/fprefect/Developer/kivy/venv/lib/python3.10/site-packages/sh.py", line 839, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_1: 

  RAN: /Users/fprefect/Developer/kivy/dist/hostpython3/bin/python setup.py build_ext -g

  STDOUT:
Running from numpy source directory.
/Users/fprefect/Developer/kivy/build/numpy/x86_64/numpy-1.24.2/setup.py:67: DeprecationWarning: 

  `numpy.distutils` is deprecated since NumPy 1.23.0, as a result
  of the deprecation of `distutils` itself. It will be removed for
  Python >= 3.12. For older Python versions it will remain present.
  It is recommended to use `setuptools < 60.0` for those Python versions.
  For more details, see:
    https://numpy.org/devdocs/reference/distutils_status_migration.html 


  import numpy.distutils.command.sdist
Cythonizing sources
Processing numpy/random/_bounded_integers.pxd.in
Processing numpy/random/_philox.pyx
warning: _philox.pyx:14:0: The 'DEF' statement is deprecated and will be removed in a future Cyt... (2281 more, please see e.stdout)

  STDERR:

```

**Screenshots**

**Additional context**

**Comment by Tan837** [comment]
Many people are facing this issue. This is a problem from the Kivy developer's end. They have changed something. I hope they fix it soon.

**Comment by pythonsus** [comment]
use this https://github.com/jswillard/kivy-ios/tree/master

**Comment by pythonsus** [comment]
it is a cython version problem since a new version came out yesterday

**Comment by misl6** [comment]
Fixed via #844
